packages/docusaurus/src/client/PendingNavigation.tsx

```diff
@@ -60,21 +63,9 @@ class PendingNavigation extends React.Component<Props, State> {
 
     // Load data while the old screen remains.
     preload(nextLocation.pathname)
-      .then(() => {
-        clientLifecyclesDispatcher.onRouteUpdate({
-          previousLocation: this.previousLocation,
-          location: nextLocation,
-        });
-        this.setState({nextRouteHasLoaded: true}, this.stopProgressBar);
-        const {hash} = nextLocation;
-        if (!hash) {
-          window.scrollTo(0, 0);
-        } else {
-          const id = decodeURIComponent(hash.substring(1));
-          const element = document.getElementById(id);
-          element?.scrollIntoView();
-        }
-      })
+      .then(() =>
+        this.setState({nextRouteHasLoaded: true}, this.stopProgressBar),
+      )
       .catch((e) => console.warn(e));
```

Comment:
This should now look much cleaner: no more DOM access in `shouldComponentUpdate` 🎉